RegExp.make.js

```diff
@@ -1065,12 +1070,26 @@ RegExp.make = (function () {
       pattern += rawLiteralPart;
       addTemplateGroups(i+1);
     }
-    const output = new RegExp(pattern, flags);
+    const output = new ctor(pattern, flags);
     output.templateGroups = templateGroups;
     return output;
   }
 
-  return make.bind(null, '' /* No flags by default */);
+  return function(x, ...values) {
+    // RegExp.make can be called in several modes.
+    // 1. RegExp.make`...undifferentiated RegExp stuff...`
+    // 2. RegExp.make('gi')`....` to specify flags
+    // 3. RegExp.make.bind(RegExpSubClass)`...` with a this value that specifies
+    //    a different constructor.
+    const xType = typeof x;
```

Comment:
Just use `typeof x` twice. Should be shorter and easier to read; also engines optimise this better when `typeof` is a direct operand of `==`/`===`.